src/Engine/GameInstance.java

```diff
@@ -283,7 +291,18 @@ public void endGame()
       army.deInitForGame(this);
     }
   }
-  
+
+  /**
+   * Gives you a random number between 0 (inclusive) and the input (exclusive)
+   * <p>Call this in GameAction.getEvents()/event constructors and nowhere else
```

Comment:
Could we enforce this access, e.g. by making it a protected static method of the `GameEvent` base class or something?
I can easily see someone misusing this call; as a member of `GameInstance`, it looks like it should be more broadly used, javadoc comment notwithstanding.

Reply:
That seems sensible. That does mean the RNG state itself has to be public, which could lead to stuff getting out of sync if you mess with it.
Of the two (exposing private state, exposing a method you shouldn't call), I am not sure which is the greater evil.
The "exposing state" probably could be done via getters/etc, but ugh.

Ideally, any solution should allow powers to directly access RNs as well, since Sturm at least needs one.

I suppose it's possible to provide public access via injection, but it seems like a lot of boilerplate: https://stackoverflow.com/a/14226229

Reply:
If it's a protected method of `GameEvent`, then it's not public, so I'm not sure what you mean. It would at least be less public than having it as a public member of `GameInstance`, which is maybe about as public and accessible as you can get.

Reply:
The *method* wouldn't be public, sure, but the members of `GameInstance` that it acts on would then have to be public.
Getting an RN from the game state seems like a pretty obviously gamestate-altering thing, and the only reason I want to restrict calls to the method is... that it alters the game state.
The API is about as clean as it can get.
Exposing the RNG-related state publicly for everyone to mess with opens you up to new categories of bad things (vs "I advanced the game's RNG accidentally")

If you mean to also push the RNG *state* into `GameEvent`, that's a non-starter IMO since I want the RNG to be consistent for a given seed and set of actions.

Reply:
Hmm, fair enough. Plus we've tried to push all action into events, so anything that would want to pull a random number should be in an event anyway.